[17.0][MIG] dms_attachment_link: Migration to 17.0

victoralmau opened the pull request:
Migration to 17.0

Please @CarlosRoca13 and @pilarvargas-tecnativa can you review it?

@Tecnativa TT50055

victoralmau committed:
[ADD] dms_attachment_link: New module

TT41512

[UPD] Update dms_attachment_link.pot

[UPD] README.rst

victoralmau committed:
[IMP] dms_attachment_link: Add searchpanel to wizard to improve file selection.

TT41512

[UPD] Update dms_attachment_link.pot

dms_attachment_link 13.0.1.1.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: dms-13.0/dms-13.0-dms_attachment_link
Translate-URL: https://translation.odoo-community.org/projects/dms-13-0/dms-13-0-dms_attachment_link/

victoralmau committed:
[MIG] dms_attachment_link: Migration to 15.0

TT41774

[UPD] Update dms_attachment_link.pot

[UPD] README.rst

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (6 of 6 strings)

Translation: dms-15.0/dms-15.0-dms_attachment_link
Translate-URL: https://translation.odoo-community.org/projects/dms-15-0/dms-15-0-dms_attachment_link/es/

[UPD] README.rst

mymage committed:
Added translation using Weblate (Italian)

Translated using Weblate (Italian)

Currently translated at 100.0% (6 of 6 strings)

Translation: dms-15.0/dms-15.0-dms_attachment_link
Translate-URL: https://translation.odoo-community.org/projects/dms-15-0/dms-15-0-dms_attachment_link/it/

DemchukM committed:
[MIG] dms_attachment_link: Migration to 16.0

Co-authored-by: DemchukM

victoralmau committed:
[MIG] dms_attachment_link: Finish migration

[UPD] Update dms_attachment_link.pot

victoralmau committed:
[IMP] dms_attachment_link: pre-commit auto fixes

victoralmau committed:
[MIG] dms_attachment_link: Migration to 17.0

TT50055

pedrobaeza commented:
/ocabot migration dms_attachment_link

chienandalu commented on dms_attachment_link/static/src/xml/chatter.xml:
```diff
@@ -1,16 +1,13 @@
                 class="o_add_dms_file_button btn btn-link"
                 type="button"
-                t-on-click.stop.prevent="attachmentBoxView._onAddDmsFile"
+                t-on-click="_onAddDmsFile"
```
Why not using the `stop.prevent` API here. It's aimed to avoid boiler plate code in the js logic

pilarvargas-tecnativa commented on dms/views/dms_file.xml:
```diff
@@ -281,7 +281,7 @@
             <kanban position="attributes">
                 <attribute name="banner_route" />
-                <attribute name="class">o_kanban_mobile</attribute>
+                <attribute name="class">o_kanban_mobile mk_file_kanban_view</attribute>
```
Shouldn't it go in a separate commit if necessary for migration?

victoralmau committed:
[IMP] dms: Add the mk_file_kanban_view class to the custom kanban view

Add the mk_file_kanban_view class to the custom kanban view so that it is
displayed correctly according to the styles set for that class (mk_file_kanban_view).

TT50055

pilarvargas-tecnativa commented:
Following the instructions in the readme, the behaviour is not the same as in previous versions. For the ‘Add DMS files’ option to appear, a file must be uploaded. In previous versions, clicking on the chatter clip already showed this option.
![dms](https://github.com/user-attachments/assets/80983bd5-9847-4044-a711-efebceae0cb1)

victoralmau commented:
Ok, I understand what you are saying, but in v17 that is the default behavior (https://github.com/odoo/odoo/blob/17.0/addons/mail/static/src/core/web/chatter.js). Should we change that behavior in this module? I think not, it should be done in another one. 

Another module (`document_url`) does something similar https://github.com/OCA/knowledge/blob/17.0/document_url/static/src/js/url.esm.js#L38 (although I don't like the replace that is defined.)

What do you think @pedrobaeza

pedrobaeza commented:
We need to possibility of adding a DMS file although no current attachment, so yes, we should add that option in this module, similar to `document_url`.

victoralmau committed:
[IMP] dms: Show Upload button in the file kanban view only if it is allowed to create

TT50055

pilarvargas-tecnativa reviewed (approved):
LGTM thanks!

pedrobaeza reviewed (approved):
/ocabot merge patch
